tcp/server.go

```diff
@@ -3,27 +3,32 @@ package tcp
 import (
 	"context"
 	"fmt"
+	"io"
 	"net"
 	"time"
 
+	"github.com/renproject/aw/handshake"
 	"github.com/renproject/aw/protocol"
 	"github.com/sirupsen/logrus"
 )
 
 type ServerOptions struct {
-	Logger  logrus.FieldLogger
-	Timeout time.Duration
+	Logger       logrus.FieldLogger
+	Timeout      time.Duration
+	SignVerifier protocol.SignVerifier
```

Comment:
Only put things into option structs if the component can set good defaults. The purpose of options is avoiding the programmer initialising a bunch of stuff that they probably don't care about. The programmer definitely needs to care about the `SignerVerifier` in this case.

Reply:
Btw, to stick with convention, it should be called `protocol.SignerVerifier` instead of `protocol.SignVerifier`.

Reply:
I have made these fixes.

Reply:
According to go stdlib, it is SignVerifier though (for ex: we have ReadWriter, ReadWriteListener, ... in the io package).